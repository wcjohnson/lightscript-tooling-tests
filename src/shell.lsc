exec = require('child_process').exec

export run(cmd) ->
  new Promise! (resolve, reject) ->
    exec! cmd, (err, stdout, stderr) ->
      if err:
        reject({err, stdout, stderr})
      else:
        resolve({stdout, stderr})

export snapshot(expect, cmd) ->
  run! cmd
    .then! ({stdout, stderr}) ->
      expect({stdout, stderr, succeeded: true, err: null}).toMatchSnapshot()
    .catch! ({err, stdout, stderr}) ->
      expect({stdout, stderr, succeeded: false, err}).toMatchSnapshot()

export successfulSnapshot(expect, cmd) ->
  run! cmd
    .then! ({stdout, stderr}) ->
      expect({stdout, stderr, succeeded: true, err: null}).toMatchSnapshot()
