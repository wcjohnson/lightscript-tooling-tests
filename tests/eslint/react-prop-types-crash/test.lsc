import { snapshot } from '../../../lib/shell'

test! "lint self", ->
  expect~snapshot! "$(npm bin)/eslint -c tests/eslint/react-prop-types-crash/linter-config.json tests/eslint/react-prop-types-crash/lintThis.lsc"
