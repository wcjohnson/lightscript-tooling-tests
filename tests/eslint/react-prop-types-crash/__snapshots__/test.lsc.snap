// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-prop-types-crash 1`] = `
Object {
  "exitStatus": 1,
  "stderr": "",
  "stdout": "
/Users/wcj/dev/lightscript/lightscript-tooling-tests/tests/eslint/react-prop-types-crash/lintThis.lsc
   81:14  error  'children' is missing in props validation  react/prop-types
   81:24  error  'section' is missing in props validation   react/prop-types
   93:23  error  'section' is missing in props validation   react/prop-types
   93:32  error  'id' is missing in props validation        react/prop-types
   93:36  error  'children' is missing in props validation  react/prop-types
  109:11  error  'code' is missing in props validation      react/prop-types
  109:17  error  'syntax' is missing in props validation    react/prop-types
  109:25  error  'width' is missing in props validation     react/prop-types
  118:16  error  'lsc' is missing in props validation       react/prop-types
  128:11  error  'code' is missing in props validation      react/prop-types

âœ– 10 problems (10 errors, 0 warnings)

",
  "succeeded": false,
}
`;
